src/controllers/api/friends.js

```diff
@@ -0,0 +1,51 @@
+const { User } = require("../../models");
+
+const createNewFriend = async (req, res) => {
+  const { userId, friendId } = req.params;
+
+  try {
+    if (userId && friendId) {
```

Comment:
We would usually see the `unhappy path` checked in your `if` blocks and then the happy path runs in the normal flow of code. This makes it easier for us to follow and eliminates those `else` blocks. Here it would look like:

```
// unhappy path
if (!userId || !friendId) {
    res.status(500).json({ success: false });
  }

// happy path
await User.findOneAndUpdate(
        { _id: userId },
        { $addToSet: { friends: friendId } },
        { new: true, runValidators: true }
      );

```